main.go

```diff
@@ -76,16 +76,30 @@ func main() {
 
 	fmt.Print("\033[s\033c")
 	for running {
-		fmt.Print("\033[u")
+		width, height, err := terminal.GetSize(int(os.Stdout.Fd()))
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to get terminal size: %s\n", err.Error())
+			return
+		}
+
 		ok := webcam.Read(&img)
 		if !ok {
 			fmt.Fprintln(os.Stderr, "Failed to read from camera")
 			return
 		}
 
-		Width, _ := strconv.Atoi(termSize[1])
-		Height := Width * img.Rows() / img.Cols()
-		cv.Resize(img, &img, image.Point{Width, Height}, 0, 0, 1)
+		// This calculates the appropriate size for the frame, maintaining
+		// the same aspect ratio, and filling the terminal window fully
+		var size image.Point
+		var termRatio = float64(height*2) / float64(width)
```

Comment:
This uses floats for two reasons: precision, and because it's really easy to hit a `divide by 0` exception when performing integer division, apparently